### PR TITLE
Add citation file

### PR DESCRIPTION
**Pull request by pcuenca**
As suggested by Khalid

### CODE REVIEW

**Comment by khalidsaifullaah** [comment]
So the issue with Github's CITATION.cff is, it generates a BibTex but it's quite broken in a sense that it messes up the authors names. For example, for the `.cff` above, we get a BibTex as the following:
```
@misc{Dayma_DALL·E_Mini_2021,
author = {Dayma, Boris and Patil, Suraj and Cuenca, Pedro and Saifullah, Khalid and Abraham, Tanishq and Lê Khắc, Phúc and Melas, Luke and Ghosh, Ritobrata},
month = {7},
title = {DALL·E Mini},
url = {https://github.com/borisdayma/dalle-mini},
year = {2021}
}
```

Now, we can either use this or go with creating our own BibTex (with `zenodo` maybe) and add it in our repo's README (seems like almost all repos do it that way).

So, what's your opinion on this? @pcuenca @borisdayma

**Comment by pcuenca** [comment]
@khalidsaifullaah what's wrong with that BibTex?

I just wanted to test what the new official citation support looked like. The nice thing about it is that it appears in a dedicated section, so I suppose it would be harder to ignore. But I don't particularly care for this too much anyway, so if you think another method is better we can close this one.

**Comment by khalidsaifullaah** [comment]
Yeah, I also think using this recent feature is a good idea, most of the users will start using it from now on. The `,` in the wrong places of `author` section are what makes the BibTex are a bit weird. But if we use our own generated BibTex, in that case, we won't be able to use this GitHub feature, which is a bummer. So, whatever you guys think, but I'm thinking let's just use this BibTex the way it is, maybe it's not that big of a deal.

**Comment by lkhphuc** [comment]
That `Dayma, Boris and Last, First name ...` is actually the correct format @khalidsaifullaah . The final output in the paper will be determine by the specific paper template.

**Comment by khalidsaifullaah** [comment]
Oh, I see, thanks for your insight @lkhphuc!
I and Boris were stressing over this naming format, but if it's the right format then I guess we can go ahead with this option!

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+date-released: 2021-07-29
+identifiers: 
+keywords: 
+  - dalle
+  - "text-to-image generation"
+  - transformer
+  - "zero-shot"
+  - JAX
+license: "Apache-2.0"
+message: "If you use this software, please cite it using these metadata."
```
Comment (khalidsaifullaah) [inline CITATION.cff:39]: Could you just include the DOI that @borisdayma has generated for our repo, you can just add it here by the following:
`doi: 10.5281/zenodo.1234`

Comment (khalidsaifullaah) [inline CITATION.cff:39]: Can we say here `Project` or `DALL·E Mini` or something else instead of `software`?
Or we must use the `software` term here...?

**Review by khalidsaifullaah — CHANGES_REQUESTED** [review]
I think we can go ahead with this option after making those minimal changes...

**Pull request merged**